include/particles/particle.tcc

```diff
@@ -1728,3 +1728,35 @@ inline double mpm::Particle<Tdim>::compute_asflip_beta(double dt) noexcept {
 
   return beta;
 }
+
+//! Rotation Function
+template <unsigned Tdim>
+void mpm::Particle<Tdim>::map_rotation_force(const VectorDim& rotation_origin,
+                                             double rotation_omega,
+                                             bool rotation_clockwise) noexcept {
+
+  // Define Angular Velocity Vector
+  Eigen::Matrix<double, 3, 1> omega_vec_3d =
+      Eigen::Matrix<double, 3, 1>::Zero();
+  // Set Z-component: Clockwise = negative Z (Right-hand rule)
+  omega_vec_3d[2] = rotation_clockwise ? -rotation_omega : rotation_omega;
+
+  // Relative Position (r) and Velocity (v) in 3D
+  Eigen::Matrix<double, 3, 1> r_p_3d = Eigen::Matrix<double, 3, 1>::Zero();
+  r_p_3d.head<Tdim>() = this->coordinates_ - rotation_origin;
+
+  Eigen::Matrix<double, 3, 1> vel_3d = Eigen::Matrix<double, 3, 1>::Zero();
+  vel_3d.head<Tdim>() = this->velocity_;
+
+  // Compute Total Rotation Force
+  const VectorDim f_total =
+      (-mass_ * (omega_vec_3d.cross(omega_vec_3d.cross(r_p_3d)) +
+                 2.0 * omega_vec_3d.cross(vel_3d)))
+          .head<Tdim>();
```

Comment:
Why f_total is VectorDim not  Eigen::Matrix<double, 3, 1>? All the other components are Eigen::Matrix<double, 3, 1>.